module/data/shared/uses-field.mjs

```diff
@@ -179,6 +179,7 @@ export default class UsesField extends SchemaField {
       }]
     }, config);
     rollConfig.subject = this;
+    rollConfig.hookNames = (config.hookNames ?? []).concat(["recharge"]);
```

Comment:
Any reason to use `concat` rather than the spread operator? It avoids an extra array creation.

```suggestion
    rollConfig.hookNames = [...(config.hookNames ?? []), "recharge"];
```